src/mono/mono/metadata/object.c

```diff
@@ -6806,6 +6806,8 @@ mono_object_handle_isinst (MonoObjectHandle obj, MonoClass *klass, MonoError *er
 {
 	error_init (error);
 
+	g_assert (klass);
```

Comment:
This assert is not particularly helpful, we will crash immediately at the line below anyway.

Reply:
on wasm we wouldn't, but i don't know if this code runs on wasm